axis.go

```diff
@@ -318,15 +326,14 @@ func (a *verticalAxis) draw(c draw.Canvas) {
 		len := a.Tick.Length
 		for _, t := range marks {
 			y := c.Y(a.Norm(t.Value))
-			if !c.ContainsY(y) {
-				continue
-			}
 			start := t.lengthOffset(len)
 			c.StrokeLine2(a.Tick.LineStyle, x+start, y, x+len, y)
 		}
 		x += len
 	}
 	c.StrokeLine2(a.LineStyle, x, c.Min.Y, x, c.Max.Y)
+
+	return a.Min, a.Max
```

Comment:
Does this mean that user-specified axes ranges are overwritten? That seems problematic.

Reply:
I'm trying to figure out ways to pass the information around that does not do this, but the engineering of the system as it stands is very tight. IMV there are a number of values that get set/passed that mean more than one thing. I'd like to try to sort that out without expanding the API too much.

Reply:
>Does this mean that user-specified axes ranges are overwritten? That seems problematic.

Only in the case that a `Ticker` returns marks that are outside the min and max given. If people don't want that, then they can write their own `Ticker` and not use `DefaultTicks`.

Reply:
I have made changes that restore the original user-specified values after using the calculated values. I think the code here can do with a clean-up, but that can happen later.